packages/snaps-rpc-methods/src/restricted/notify.ts

```diff
@@ -173,13 +173,27 @@ export function getValidatedParams(params: unknown): NotifyParams {
     });
   }
 
+  const isNotString = !message || typeof message !== 'string';
   // Set to the max message length on a Mac notification for now.
-  if (!message || typeof message !== 'string' || message.length >= 50) {
+  if (
+    isNotString ||
```

Comment:
This is somewhat flawed since specifying a non-string will get you the first error message, telling you the limit is 50 chars, even if the limit is 500 chars for your selected notification type

Reply:
Fixed: [6e724dc](https://github.com/MetaMask/snaps/pull/2684/commits/6e724dc7c3c6f25565e5bb55a549787ca9e5f9cb)